### PR TITLE
[MI-2814] Done the review fixes of github PR #636

### PR DESCRIPTION
**Pull request by Nityanand13**
PR [#636](https://github.com/mattermost/mattermost-plugin-github/pull/636)

### CODE REVIEW

File: webapp/src/components/modals/create_update_issue/create_update_issue.jsx [hunk 1]
```diff
@@ -46,36 +48,53 @@ export default class CreateOrUpdateIssueModal extends PureComponent {
...
+    updateState(issueInfo) {
+        var {channel_id, title, description, assignees, labels, milestone_title, milestone_number, repo_full_name} = issueInfo ?? {};
+        if (!assignees) {
+            assignees = [];
+        }
+        if (!labels) {
+            labels = [];
+        }
+
+        this.setState({assignees});
```
Comment (Kshitij-Katiyar) [inline webapp/src/components/modals/create_update_issue/create_update_issue.jsx]: i think you can set all the states in a single line

File: webapp/src/components/modals/create_update_issue/create_update_issue.jsx [hunk 2]
```diff
@@ -46,36 +48,53 @@ export default class CreateOrUpdateIssueModal extends PureComponent {
         this.validator = new Validator();
     }
 
+    getIssueInfo = async () => {
+        const issueInfo = await this.props.issueInfo(this.props.messageData);
+        return issueInfo;
+    }
+
+    updateState(issueInfo) {
+        var {channel_id, title, description, assignees, labels, milestone_title, milestone_number, repo_full_name} = issueInfo ?? {};
```
Comment (Kshitij-Katiyar) [inline webapp/src/components/modals/create_update_issue/create_update_issue.jsx]: are you updating any of these variable if not use const

Comment (Kshitij-Katiyar) [inline webapp/src/components/modals/create_update_issue/create_update_issue.jsx]: any reason that for issueInfo you have used snake case but casing is different on line 57

Comment (Nityanand13) [reply webapp/src/components/modals/create_update_issue/create_update_issue.jsx]: Discussed

File: webapp/src/components/modals/attach_comment_to_issue/attach_comment_to_issue.jsx [hunk 3]
```diff
@@ -106,9 +106,17 @@ export default class AttachIssueModal extends PureComponent {
         });
     };
 
+    /* eslint-disable react/no-did-update-set-state*/
```
Comment (Kshitij-Katiyar) [inline webapp/src/components/modals/attach_comment_to_issue/attach_comment_to_issue.jsx]: use next or same line disable lint command use it for 112 line

File: webapp/src/components/modals/create_update_issue/create_update_issue.jsx [hunk 4]
```diff
@@ -252,6 +252,7 @@ export default class CreateOrUpdateIssueModal extends PureComponent {
                     onChange={this.handleIssueTitleChange}
                 />
                 {issueTitleValidationError}
+
```
Comment (Kshitij-Katiyar) [inline webapp/src/components/modals/create_update_issue/create_update_issue.jsx:252]: why this change ?

**Review by Kshitij-Katiyar — APPROVED** [review]

File: webapp/src/components/modals/create_update_issue/create_update_issue.jsx (hunk 4 above)
Comment (Nityanand13) [reply webapp/src/components/modals/create_update_issue/create_update_issue.jsx:252]: For consistency

File: server/plugin/api.go [hunk 5]
```diff
@@ -97,9 +97,7 @@ func (p *Plugin) initializeAPI() {
 	apiRouter.HandleFunc("/create_issue", p.checkAuth(p.attachUserContext(p.createIssue), ResponseTypePlain)).Methods(http.MethodPost)
 	apiRouter.HandleFunc("/close_or_reopen_issue", p.checkAuth(p.attachUserContext(p.closeOrReopenIssue), ResponseTypePlain)).Methods(http.MethodPost)
 	apiRouter.HandleFunc("/update_issue", p.checkAuth(p.attachUserContext(p.updateIssue), ResponseTypePlain)).Methods(http.MethodPost)
-	apiRouter.HandleFunc("/edit_issue_modal", p.checkAuth(p.attachUserContext(p.openIssueEditModal), ResponseTypePlain)).Methods(http.MethodPost)
-	apiRouter.HandleFunc("/close_reopen_issue_modal", p.checkAuth(p.attachUserContext(p.openCloseOrReopenIssueModal), ResponseTypePlain)).Methods(http.MethodPost)
-	apiRouter.HandleFunc("/attach_comment_issue_modal", p.checkAuth(p.attachUserContext(p.openAttachCommentIssueModal), ResponseTypePlain)).Methods(http.MethodPost)
+	apiRouter.HandleFunc("/issue_info", p.checkAuth(p.attachUserContext(p.getIssueInfo), ResponseTypePlain)).Methods(http.MethodPost)
```
Comment (manojmalik20) [inline server/plugin/api.go]: Why is the method `POST` when we are getting the issue info?

File: webapp/src/actions/index.js [hunk 6]
```diff
@@ -211,47 +211,11 @@ export function getMilestoneOptions(repo) {
     };
 }
 
-export function attachCommentIssueModal(payload) {
+export function issueInfo(payload) {
```
Comment (manojmalik20) [inline webapp/src/actions/index.js]:
```suggestion
export function getIssueInfo(payload) {
```

File: webapp/src/client/client.js [hunk 7]
```diff
@@ -7,16 +7,8 @@ import {ClientError} from 'mattermost-redux/client/client4';
...
-        return this.doPost(`${this.url}/edit_issue_modal`, payload);
-    }
-
-    closeOrReopenIssueModal = async (payload) => {
-        return this.doPost(`${this.url}/close_reopen_issue_modal`, payload);
-    }
-
-    attachCommentIssueModal = async (payload) => {
-        return this.doPost(`${this.url}/attach_comment_issue_modal`, payload);
+    issueInfo = async (payload) => {
```
Comment (manojmalik20) [inline webapp/src/client/client.js]:
```suggestion
    getIssueInfo = async (payload) => {
```

File: webapp/src/components/modals/create_update_issue/create_update_issue.jsx (hunk 2 above)
Comment (manojmalik20) [reply webapp/src/components/modals/create_update_issue/create_update_issue.jsx]: Use `const`.

File: webapp/src/components/modals/create_update_issue/create_update_issue.jsx [hunk 8]
```diff
@@ -46,36 +48,53 @@ export default class CreateOrUpdateIssueModal extends PureComponent {
...
+    getIssueInfo = async () => {
+        const issueInfo = await this.props.issueInfo(this.props.messageData);
+        return issueInfo;
+    }
+
+    updateState(issueInfo) {
+        var {channel_id, title, description, assignees, labels, milestone_title, milestone_number, repo_full_name} = issueInfo ?? {};
+        if (!assignees) {
+            assignees = [];
+        }
```
Comment (manojmalik20) [inline webapp/src/components/modals/create_update_issue/create_update_issue.jsx]:
```suggestion
        const assignees = issueInfo?.assignees ?? [];
```

File: webapp/src/components/modals/create_update_issue/create_update_issue.jsx [hunk 9]
```diff
@@ -46,36 +48,53 @@ export default class CreateOrUpdateIssueModal extends PureComponent {
...
-            this.setState({milestone: {
-                value: milestone_number,
-                label: milestone_title,
-            },
-            repo: {
-                name: repo_full_name,
-            },
-            issueDescription: description,
-            issueTitle: title.substring(0, MAX_TITLE_LENGTH)});
+        if (this.props.messageData && this.props.messageData.repo_owner && !prevProps.visible && this.props.visible) {
```
Comment (manojmalik20) [inline webapp/src/components/modals/create_update_issue/create_update_issue.jsx]:
```suggestion
        if (this.props.messageData?.repo_owner && !prevProps.visible && this.props.visible) {
```

**Review by manojmalik20 — APPROVED** [review]
